### The data analysed is from this Kaggle data set:
https://www.kaggle.com/ntnu-testimon/paysim1



In [1]:
import pandas as pd

In [2]:
#load data
df = pd.read_csv('financial.csv')

### Types of Transactions

This has a look at which types of transactions are more likely to be frauduent.

Results: cash out and transfer are the only types to show fraudulent activity. Therefore we can assume that transactions of this type are more likely to be fraudulent. If we take a sanity check this makes sense as it would be unusual to fraudulently add money to someones account.

In [3]:
#number of occurances of each type split by isFraud/not
real_trans_groups = df[df['isFraud'] == 0].groupby('type')
real_trans_counts = real_trans_groups.count()
print(real_trans_counts)
fake_trans_groups = df[df['isFraud'] == 1].groupby('type')
fake_trans_counts = fake_trans_groups.count()
print(fake_trans_counts)

             step   amount  nameOrig  oldbalanceOrg  newbalanceOrig  nameDest  \
type                                                                            
CASH_IN   1399284  1399284   1399284        1399284         1399284   1399284   
CASH_OUT  2233384  2233384   2233384        2233384         2233384   2233384   
DEBIT       41432    41432     41432          41432           41432     41432   
PAYMENT   2151495  2151495   2151495        2151495         2151495   2151495   
TRANSFER   528812   528812    528812         528812          528812    528812   

          oldbalanceDest  newbalanceDest  isFraud  isFlaggedFraud  
type                                                               
CASH_IN          1399284         1399284  1399284         1399284  
CASH_OUT         2233384         2233384  2233384         2233384  
DEBIT              41432           41432    41432           41432  
PAYMENT          2151495         2151495  2151495         2151495  
TRANSFER          528812

### Transaction amounts

We now look at the data to see if larger transactions are more likely to be fraudulent. First lets find out how many fraudulent transactions there are in the data set. 

In [5]:
#number of frauds
total_fraud = df.isFraud[df['isFraud']== 1].count()
print(total_fraud)

8213


The number of frauds in the data is 8213, next we will look at the number of fradulent transactions 
for different ammounts the higher you go the number of true transactions gets lower really quickly so this seems to be an indicator.

In [6]:
frauds = df[df['isFraud']==1]
high_amount =frauds.amount[frauds['amount']>80000].count()
print(high_amount)
#percentage of frauds 
percent = (high_amount/total_fraud)*100
percent = round(percent,2)
print(percent)

6743
82.1


There are 8161 fraudulent transactions for 500 > OR 99.37%, so what happens if we put a value like 30 (the most that can be used on a contactless card without using the pin).

In [7]:
low_amount = frauds.amount[frauds['amount'] <=30].count()
print(low_amount)
low_percent = low_amount/total_fraud*100
print(low_percent)

16
0.194813101181


From these numbers it would look more probable that larger transactions are likely to be fraudulent. Lets check the amounts of non-fraudulent transactions.

In [8]:
total_real_trans = df.isFraud[df['isFraud']==0].count()
print(total_real_trans)
real_trans = df[df['isFraud']==0]
high_amount_real = real_trans.amount[real_trans['amount']>80000].count()
print(high_amount_real)
high_percentage = high_amount_real/total_real_trans*100
print(high_percentage)

6354407
3102973
48.8318264789


There are no timestamps in the data -set back but then could be randomly generated?

In [10]:
#check new balance for origin is fraud always 0 newbalanceOrig
cash_out= frauds.newbalanceOrig[frauds['isFraud']==1].count()
print(cash_out)

8213


check the real transactions

In [12]:
cash_out_real = real_trans.newbalanceOrig[real_trans['isFraud']==0].count()
print(cash_out_real)

6354407


Look at these results all of the fraudulent transactions have a bank account with 0 left after the transaction this is not the case with real transactions.